File: FE/AL-homepage/src/types/testResult.ts
export interface TestResult {
  test_id: number
  student_id: number
  score: number
  result_id: number
}

export interface CreateTestResultRequest {
  test_id: number
  student_id: number
  score: number
}

export interface UpdateTestResultRequest {
  test_id?: number
  student_id?: number
  score?: number
}

export interface GetAllTestResultsParams {
  skip?: number
  limit?: number
  test_id?: number
  student_id?: number
}

File: BE/avalon/app/models/test_result.py
from sqlalchemy import Column, Integer, ForeignKey
from .base import Base

class TestResult(Base):
    __tablename__ = "test_results"
    result_id = Column(Integer, primary_key=True, autoincrement=True)
    test_id = Column(Integer, ForeignKey("test.test_id", ondelete="CASCADE"), nullable=False)
    student_id = Column(Integer, ForeignKey("students.student_id"), nullable=False)
    score = Column(Integer, nullable=False)